### PR TITLE
18639 amalgamation validation

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#18639

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/tests/unit/services/filings/validations/test_amalgamation_application.py [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+
+    # validate outcomes
+    if test_status == 'FAIL':
+        assert expected_code == err.code
+        assert expected_msg == err.msg[0]['error']
+    else:
+        assert not err
+
+
+def test_amalgamating_businesses(mocker, app, session):
```
Comment (vysakh-menon-aot) [inline legal-api/tests/unit/services/filings/validations/test_amalgamation_application.py:642]: Updating rest of the unit tests

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        identifier = amalgamating_business_json.get('identifier')
+        foreign_legal_name = amalgamating_business_json.get('legalName')
+        is_foreign_business = bool(foreign_legal_name)
+        amalgamating_business = amalgamating_businesses.get(identifier)
+
+        if amalgamating_business and _has_future_effective_filing(amalgamating_business):
+            msg.append({
+                'error': f'{identifier} has a future effective filing',
+                'path': amalgamating_businesses_path
+            })
```
Comment (argush3) [inline legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:118]: so validation fails on any future effective filing on a TING?

**Comment by argush3** [comment]
looks good, just the one question.

**Review by argush3 — APPROVED** [review]

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py (hunk 2 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:118]: yes. is in't that how it should be?

Comment (argush3) [reply legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:118]: I haven't looked at that validation requirements document. 

Just wanted to confirm.

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:118]: as per the document it is

**Review by kzdev420 — APPROVED** [review]
LGTM

**Pull request merged**